write-fonts/src/lib.rs

```diff
@@ -1,4 +1,100 @@
-//! Raw types for compiling opentype tables
+//! Writing and modifying OpenType tables
+//!
+//! This crate provides a collection of types correlating to what is described
+//! in the [OpenType spec][spec], along with the logic to serialize these types
+//! into binary font tables. It is a companion to [`read-fonts`], which provides
+//! efficient zero-allocation parsing of these types. It is intended to be used
+//! as the basis for font engineering tools such as compilers.
+//!
+//! ## 'write' versus 'read' types
+//!
+//! Both `write-fonts` and [`read-fonts`] make heavy use of code generation, and
+//! they have a similar structure, where a `tables` module contains a submodule for
+//! each supported [table][table-directory], and that module contains items for
+//! each table, record, flagset or enum described in the spec. This means that
```

Comment:
This is a great description of the code layout. Can we add this to the `read-fonts` docs as well? I think the duplication is worth it.